identity/opendj/opendj_backend.py

```diff
@@ -0,0 +1,349 @@
+            - Either password or passwordfile is needed.
+        required: false
+    name:
+        description:
+            - The name of the backend to create.
+        required: true
+    bind_dn:
+        description:
+            - The username to connect to OpenDJ.
+        required: false
+        default: cn=Directory Manager
+    base_dn:
+        description:
+            - The base DN. Example dc=example,dc=com.
+        required: True
+    type:
+        description:
+            - The type of backend you want to use for OpenDJ.
+        required: false
+        choices: ["je", "pdb"]
+        default: "je"
+    enabled:
+        description:
+            - If the backend needs to be enabled when the backend is created.
+        required: false
+        choices: ['True', 'False']
+        default: "True"
+    state:
+        description:
+            - If configuration needs to be added/deleted
+        required: false
+        choices: ['present', 'absent']
+        default: "present"
+'''
+
+EXAMPLES = '''
+    - name: "Create Pizza Backend"
+      action: opendj_backend
+              hostname=localhost
+              port=4444
+              password=password
+              name=myPizzaBackend
+              bind_dn="cn=Directory Manager"
+              base_dn=dc=ansible,dc=example,dc=com
+              enabled=true
+
+    - name: "Delete Pizza Backend"
+      action: opendj_backend
+              hostname=localhost
+              port=4444
+              password=password
+              name=myPizzaBackend
+              bind_dn="cn=Directory Manager"
+              base_dn=dc=ansible,dc=example,dc=com
+              type=pdb
+              enabled=true
+              state=absent
+
+'''
+
+RETURN = '''
+
+'''
+
+import subprocess
+import time
+from tempfile import mkstemp
+import os
+import re
+
+
+class Backend(object):
+    def __init__(self, module):
+        self._module = module
+
+    def get_backend(self, opendj_bindir, hostname, port, password_method, bind_dn):
+        my_command = [
+            opendj_bindir + '/dsconfig',
+            'list-backends',
+            '-h', hostname,
+            '-p', str(port),
+            '--bindDN', bind_dn,
+            '-X'
+        ] + password_method
+        process = subprocess.Popen(my_command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = process.communicate()
+
+        if process.returncode == 0:
+            return stdout
+        else:
+            return self._module.fail_json(msg="Error message while getting list backends: " + str(stderr))
+
+    def create_backend(self, opendj_bindir, hostname, port, password_method, name, base_dn, bind_dn, type, enabled):
+        my_command = [
+            opendj_bindir + '/dsconfig',
+            'create-backend',
+            '--backend-name', name,
+            '--set', 'base-dn:' + base_dn,
+            '--set', 'enabled:' + str(enable).lower(),
```

Comment:
oh, this is embarrassing. Yes, fixed I checked the file again to see if I missed more. luckily non other found. :-)